common/tier4_autoware_utils/CMakeLists.txt

```diff
@@ -19,6 +19,11 @@ ament_auto_add_library(tier4_autoware_utils SHARED
   src/planning/planning_marker_helper.cpp
 )
 
+# workaround to allow deprecated header to build on both galactic and rolling
+target_compile_options(tier4_autoware_utils PUBLIC
```

Comment:
Instead of ignoring errors, I'd pass the version of the `tf2_geometry_msgs` package to the code so that we can choose what header to use.

Like this:

```cmake
target_compile_definitions(${PROJECT_NAME}
  PRIVATE
    TF2_GEOMETRY_MSGS_VERSION_MAJOR=${tf2_geometry_msgs_VERSION_MAJOR}
    TF2_GEOMETRY_MSGS_VERSION_MINOR=${tf2_geometry_msgs_VERSION_MINOR}
    TF2_GEOMETRY_MSGS_VERSION_PATCH=${tf2_geometry_msgs_VERSION_PATCH}
)
```

then in the code, we can do this:

```c++
#define TF2_GEOMETRY_MSGS_VERSION_GTE(major, minor, patch) ( \
       major < TF2_GEOMETRY_MSGS_VERSION_MAJOR ? true        \
     : major > TF2_GEOMETRY_MSGS_VERSION_MAJOR ? false       \
     : minor < TF2_GEOMETRY_MSGS_VERSION_MINOR ? true        \
     : minor > TF2_GEOMETRY_MSGS_VERSION_MINOR ? false       \
     : patch < TF2_GEOMETRY_MSGS_VERSION_PATCH ? true        \
     : patch > TF2_GEOMETRY_MSGS_VERSION_PATCH ? false       \
     : true)

...

#if TF2_GEOMETRY_MSGS_VERSION_GTE(X,Y,Z) // X, Y, Z is the version in Rolling
#include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
#else
#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
#endif
```